ts/features/wallet/onboarding/bancomat/screens/bpd/ActivateBpdOnNewPaymentMethodScreen.tsx

```diff
@@ -26,8 +34,12 @@ const loadLocales = () => ({
   continueStr: I18n.t("global.buttons.continue")
 });
 
-const ActivateBpdOnNewBancomatScreen: React.FunctionComponent<Props> = props => {
+const ActivateBpdOnNewPaymentMethodScreen: React.FunctionComponent<Props> = props => {
   const { headerTitle, title, body1, body2, skip, continueStr } = loadLocales();
+  // show new added methods from navigation params
+  // if they aren't, read the new bancomat added from the store
+  const newMethodsAdded =
+    props.navigation.getParam("newAddedMethods") ?? props.newBancomat;
```

Comment:
In order to be ready for the next payments methods, would be nice to transform `ActivateBpdOnNewBancomatScreen` -> `ActivateBpdOnNewPaymentMethodScreen`, a component that accepts as props: `paymentMethods: ReadonlyArray<PatchedWalletV2>` and than add:
```typescript

const ActivateBpdOnNewBancomatScreen: React.FunctionComponent<Props> = props => (
<ActivateBpdOnNewPaymentMethodScreen paymentMethods={props.newBancomat} />
);

const mapStateToProps = (state: GlobalState) => ({
  newBancomat: onboardingBancomatAddedPansSelector(state)
});

```

```typescript

const ActivateBpdOnNewCreditCardScreen: React.FunctionComponent<Props> = props => (
<ActivateBpdOnNewPaymentMethodScreen paymentMethods={props.navigation.getParam("newAddedMethods")} />
);

```

In this way we can reuse the `ActivateBpdOnNewPaymentMethodScreen` also for future payments type that can be read from store or from navigation props.
What you think?

Reply:
update within https://github.com/pagopa/io-app/pull/2399/commits/b40f05afe628ce3b423d88902aca4fa40f212591